### DISCUSSION TITLE
Correct way to consume GitVersion injected version metadata

### DISCUSSION THREAD

**Discussion by joaosantana** [GitTools/GitVersion#2811] [Q&A]
I want to show my app's semver in a TextBlock. For that, I have this method:

```csharp
private string GetVersionDescription()
    {
        // Semantic version provided by GitVersion
        var appName = "AppDisplayName".GetLocalized();
        var appVersion = GitVersionInformation.SemVer;

        return $"{appName} - {appVersion}";
    }
```

When I compile, however, I get this error:

`The "CompileXaml" task returned false but did not log an error.`

What I'm doing wrong and what's the correct way to consume the injected metadata?

Thanks your your time.

**Comment by asbjornu** [comment]
Try retrieving the version number with reflection instead, perhaps?
